utils/frame/try-runtime/cli/src/commands/on_runtime_upgrade.rs

```diff
@@ -29,12 +29,37 @@ pub struct OnRuntimeUpgradeCmd {
 	#[command(subcommand)]
 	pub state: State,
 
-	/// Execute `try_state`, `pre_upgrade` and `post_upgrade` checks as well.
+	/// Select which optional checks to perform.
 	///
-	/// This will perform more checks, but it will also makes the reported PoV/Weight be
-	/// inaccurate.
-	#[clap(long)]
-	pub checks: bool,
+	/// Performing any checks will potentially invalidate the measured PoV/Weight.
+	#[clap(long, value_enum, default_value_t = UpgradeCheckSelect::All)]
+	pub checks: UpgradeCheckSelect,
+}
+
+/// This is an adapter for [`frame_try_runtime::UpgradeCheckSelect`] since that does not implement
+/// `clap::ValueEnum`.
+#[derive(clap::ValueEnum, Debug, Clone, Copy)]
+#[value(rename_all = "kebab-case")]
+pub enum UpgradeCheckSelect {
```

Comment:
Seems like this is an alternative to implementing `FromStr` on a raw enum as I did for `enum Select`. Might prefer the latter, as you won't need to re-define the enum twice. Nonetheless, above all, let's keep them consistent and use one paradigm, regardless of which one is better.

Reply:
Also, we can then implement `FromStr` such that `--checks` equals `--checks all` (with a warning) and keep it backwards compatible.

Reply:
I tried it to be backwards compatible, but if I do `on-runtime-upgrade--checks live --uri …` then Clap tries to parse the `live` as `checks`, which obviously does not work :facepalm:  

I had the `FromStr` first but the downsides where that: manual documentation and possible incorrect impl of `FromStr`.  
Then opted for two Enums in e29538c1a79d1711b43addc9400d871f6aa32844, anyway I can revert that.

Reply:
> I tried it to be backwards compatible, but if I do on-runtime-upgrade--checks live --uri … then Clap tries to parse the live as checks, which obviously does not work 🤦

Might be quite hacky, but given the layout, if you parse `--checks [live|snap]` it would mean that it was meant to be `--checks all [live-snap]` etc.

Reply:
Hm then I would have to *un-parse* that thing somehow to parse it again as sub-command, I dont think that works nicely in Clap.  
Or: We rename the arg to something like `--select-checks` and keep the old `--checks` behavious.

Reply:
https://github.com/clap-rs/clap/blob/a87b55900d6c60e9bf76b61fb31efc07078f147b/examples/git-derive.rs#L38 `default_missing_value` should help for making `--checks` accept no value?

Reply:
Ah this works because it requires `=`, smart… updated it.